src/runner/pythonrunner.cpp

```diff
@@ -724,9 +503,12 @@ int getArgCount(PyObject *object) {
   return result;
 }
 
+template <typename>
+struct Functor_converter;
+
```

Comment:
Why do we need the separate declaration?

Reply:
You need separate declarations to have the `Ret(Args... )` specialization, otherwize, you have to extract `R` and `Args...` from `T` which, as far as I know, is not possible (or at least would be much trouble).

Reply:
Why does it need to be a specialisation rather than the only version? We're not instantiating any other version.

Reply:
You cannot directly declare `Functor_converter<R(Args...)>`, you have to use a specialization. If you know a way to do this, feel free to tell me and I'll do the modification immediately.

You can check, e.g., `libstdc++` and see that's what they do for `std::function`, and they also have only one version:

https://github.com/gcc-mirror/gcc/blob/master/libstdc%2B%2B-v3/include/bits/std_function.h#L113

https://github.com/gcc-mirror/gcc/blob/master/libstdc%2B%2B-v3/include/bits/std_function.h#L303

Reply:
Fair enough. I don't see any obvious reason why templates would need to work like that, but sometimes templates make no sense.